src/Store/CookieStore.php

```diff
@@ -79,21 +140,80 @@ public function delete($key)
     {
         $key_name = $this->getCookieName($key);
         unset($_COOKIE[$key_name]);
-        $this->setCookie( $key_name );
+        $this->setCookie( $key_name, '', 0 );
+
+        // If we set a legacy fallback value, remove that as well.
+        if ($this->legacySameSiteNone) {
+            unset($_COOKIE['_'.$key_name]);
+            $this->setCookie( '_'.$key_name, '', 0 );
+        }
     }
 
     /**
-     * Set or delete a cookie.
+     * Build the header to use when setting SameSite cookies.
      *
-     * @param string  $name    Cookie name to set.
-     * @param string  $value   Cookie value.
-     * @param integer $expires Cookie expiration.
+     * @param string  $name   Cookie name.
+     * @param string  $value  Cookie value.
+     * @param integer $expire Cookie expiration timecode.
+     *
+     * @return string
+     */
+    protected function getSameSiteCookieHeader(string $name, string $value, int $expire) : string
+    {
+        $date = new \Datetime();
+        $date->setTimestamp($expire)
+            ->setTimezone(new \DateTimeZone('GMT'));
+
+        return sprintf(
+            'Set-Cookie: %s=%s; path=/; expires=%s; HttpOnly; SameSite=%s%s',
+            $name,
+            $value,
```

Comment:
Do we need to escape name and value here to prevent `;` breaking out?

Reply:
Very good catch! That was missed in the core `setcookie()` method and fixed in PHP 7.3!

https://github.com/php/php-src/commit/5cb825df7251aeb28b297f071c35b227a3949f01